explorer-api/src/Explorer/Web/Api/Legacy/AddressSummary.hs

```diff
@@ -159,10 +159,9 @@ queryCTxBriefs xs = do
 queryTxInputs :: MonadIO m => [TxId] -> SqlPersistT m [(TxId, [CTxAddressBrief])]
 queryTxInputs txids = do
     rows <- select . distinct . from $ \(tx `InnerJoin` txIn `InnerJoin` txOut `InnerJoin` txInTx) -> do
-              on (txInTx ^. TxId ==. txOut ^. TxOutTxId)
-              on (txIn ^. TxInTxOutId ==. txOut ^. TxOutTxId
-                  &&. txIn ^. TxInTxOutIndex ==. txOut ^. TxOutIndex)
               on (tx ^. TxId ==. txIn ^. TxInTxInId)
+              on (txIn ^. TxInTxOutId ==. txOut ^. TxOutTxId &&. txIn ^. TxInTxOutIndex ==. txOut ^. TxOutIndex)
```

Comment:
:confounded:

Reply:
?

Reply:
I get operator blindness here :cry:

Reply:
Yes. Especially when it's actually the main reason behind the 500 internal runtime errors we were getting. Some `on` clauses wrongly specified and quite hard to catch because everything looks alike and isn't really readable :/ ...

Reply:
It would look clearer with normal record accessors instead of the lens operator.
And there is probably a way to get better field names out of persistent.

Reply:
@rvl I agree. So far I am mostly only doing "least effort maintenance" on cardano-rest and I am not really willing to put any extra effort on refactoring this code. Rather, I want to throw it in the garbage and recommend better solutions like cardano-graphql or cardano-rosetta.